src/main/kotlin/com/model/Reservation.kt

```diff
@@ -0,0 +1,38 @@
+package com.model
+
+import java.math.BigDecimal
+import java.time.LocalDateTime
+import javax.persistence.Entity
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Entity
+@Table(name = "reservations")
+data class Reservation(
+
+        @Id
+        @GeneratedValue(strategy = GenerationType.AUTO)
+        val id: Long = 0,
+
+        @NotNull
+        @Enumerated(EnumType.STRING)
+        val driverType: DriverType,
+
+        @Size(min = 4, max = 8)
+        val carLicensePlate: String,
+
+        var startTime: LocalDateTime,
+
+        var endTime: LocalDateTime? = null,
```

Comment:
czemu nie val (w tym i w innych przypadkach w tej klasie)?

Reply:
Ponieważ po rozpoczęciu rezerwacji endTime nie istnieje, myślałem nad przyjęciem konwencji że domyślna wartość to LocalDateTime.MIN ale następnie robienie query do bazy z wartością LocalDateTime.MIN zamiast null wydawało mi się gorsze.

Reply:
Jeśli stworzylibyśmy obiekt niemutowalny to za każdym updejtem musielibyśmy tworzyć nową encję. Na początku wydawało mi się to mało intuicyjnę że byśmy usuwali starą encję i tworzyli nową w bazie z każdym updejtem. Aczkolwiek bo przedyskutowaniu tego na grupie open sourcowej jednak zgodzę się że niemutowalne encję są dobrą koncepcją. Cytując "w praktyce runtime "wie" ze ten obiekt juz nie bedzie uzywany a nowy rozni sie od starego 1 polem
wiec runtime "mutuje" ten obiekt
ale trik polega na tym, ze jako programista masz pelne prawo udawac, ze o tym nie wiesz"